clients/spark/core/src/main/scala/io/treeverse/clients/Exporter.scala

```diff
@@ -133,31 +134,32 @@ object Exporter {
 
     val dstFS = dstPath.getFileSystem(conf)
 
-      action match {
-        case "delete" => {
-          try {
-            dstFS.delete(dstPath, false)
-            null
-          } catch {
-            case e : (IOException) =>  s"Unable to delete file ${dstPath.toString}: ${e.toString}"
-          }
+    action match {
+      case "delete" => {
+        try {
+          dstFS.delete(dstPath, false)
+          null
+        } catch {
+          case e : (IOException) =>  ExportStatus(dstPath.toString, success = false, s"Unable to delete file ${dstPath.toString}: ${e.toString}")
         }
+      }
 
-        case "copy" =>{
-          try {
-            org.apache.hadoop.fs.FileUtil.copy(
-              srcPath.getFileSystem(conf),
-              srcPath,
-              dstFS,
-              dstPath,
-              false,
-              conf)
-            null
-          } catch {
-            case e : (IOException) =>  s"Unable to copy file ${dstPath.toString} from source ${srcPath.toString}: ${e.toString}"
-          }
+      case "copy" =>{
+        try {
+          org.apache.hadoop.fs.FileUtil.copy(
+            srcPath.getFileSystem(conf),
+            srcPath,
+            dstFS,
+            dstPath,
+            false,
+            conf)
+          null
+        } catch  {
+          case e : (FileNotFoundException) =>  ExportStatus(dstPath.toString, success = true, s"Unable to copy file ${dstPath.toString} from source ${srcPath.toString} since source file is missing: ${e.toString}")
+          case e : (IOException) =>  ExportStatus(dstPath.toString, success = false, s"Unable to copy file ${dstPath.toString} from source ${srcPath.toString}: ${e.toString}")
```

Comment:
Do you really need all those toString calls when interpolating strings? (If so, possibly a reason to use % formatting)

Reply:
Done.